### PR TITLE
`azurerm_pim_eligible_role_assignment`: use timeout from `ctx` instead of a fix value

### LINKED ISSUE #22588: azurerm_pim_active_role_assignment failing when the azurerm_role_definition is provisioned at a different subscription
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.2

### AzureRM Provider Version

3.65

### Affected Resource(s)/Data Source(s)

azurerm_pim_active_role_assignment, azurerm_pim_eligible_role_assignment

### Terraform Configuration Files

```hcl
data "azurerm_subscription" "primary" {
}
resource "azurerm_role_definition" "example" {
  name        = "my-custom-role"
  scope       = data.azurerm_subscription.primary.id
  description = "This is a custom role created via Terraform"

  permissions {
    actions     = ["*"]
    not_actions = []
    data_actions = ["*"]
    not_data_actions = []
  }

  assignable_scopes = [
    data.azurerm_subscription.primary.id,
    "/subscriptions/<subscriptionID2>"
  ]

}

resource "azurerm_pim_active_role_assignment" "always_active_azure_role_assignment1" {
  scope              = data.azurerm_subscription.primary.id
  role_definition_id = azurerm_role_definition.example.role_definition_resource_id
  principal_id       = "<principal_id>"
  justification      = "Test1"


  schedule {
    expiration {
      duration_hours = 10
    }
  }

}

resource "azurerm_pim_active_role_assignment" "always_active_azure_role_assignment2" {
  scope              = "/subscriptions/<subscriptionID2>"
  role_definition_id = azurerm_role_definition.example.role_definition_resource_id
  principal_id       = "<principal_id>"
  justification      = "Test2"


  schedule {
    expiration {
      duration_hours = 10
    }
  }

}
```

### Debug Output/Panic Output

```shell
Error: waiting for Role Management Policy: (Principal Id "<principal_id>" / Scope "/subscriptions/<subscriptionID2>" / Role Definition Id "<data.azurerm_subscription.primary.id>/providers/Microsoft.Authorization/roleDefinitions/<role_definition_id>") to become found: timeout while waiting for state to become 'Found' (last state: 'Missing', timeout: 5m0s)
```

### Expected Behaviour

_No response_

### Actual Behaviour

Error: waiting for Role Management Policy: (Principal Id "<principal_id>" / Scope "/subscriptions/<subscriptionID2>" / Role Definition Id "<data.azurerm_subscription.primary.id>/providers/Microsoft.Authorization/roleDefinitions/<role_definition_id>") to become found: timeout while waiting for state to become 'Found' (last state: 'Missing', timeout: 5m0s)

This error occurred when provisioning azurerm_pim_active_role_assignment in subscriptionID2, even though it is included in the assignable_scope. azurerm_pim_active_role_assignment in subscriptionID2 is created when I checked from Azure Portal

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by wuxu92**
may fixes #22588

### CODE REVIEW

File: internal/services/authorization/pim_eligible_role_assignment_resource.go [hunk 1]
```diff
@@ -653,3 +653,10 @@ func deleteEligibilityRoleAssignmentSchedule(ctx context.Context, client *roleel
 		return result, "Deleted", nil
 	}
 }
+
+func waitTimeoutFromCtx(ctx context.Context, defaultValue time.Duration) time.Duration {
```
Comment (tombuildsstuff) [inline internal/services/authorization/pim_eligible_role_assignment_resource.go:657]: this default value will never get used - `ctx`'s should _always_ be wrapped by a timeout, so can we remove this method and instead use:

```go
deadline, ok := ctx.Deadline()
if !ok {
  return fmt.Errorf("internal-error: context had no deadline")
}

// ...
Timeout: time.Until(deadline),
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Left a comment inline, since this default value will never get used

File: internal/services/authorization/pim_eligible_role_assignment_resource.go (hunk 1 above)
Comment (wuxu92) [reply internal/services/authorization/pim_eligible_role_assignment_resource.go]: that makes seanse. updated in this way.

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for this @wuxu92

**Pull request merged**